embed/config.go

```diff
@@ -346,14 +331,27 @@ func (cfg Config) ElectionTicks() int { return int(cfg.ElectionMs / cfg.TickMs)
 
 // IsDefaultHost returns the default hostname, if used, and the error, if any,
 // from getting the machine's default host.
-func (cfg Config) IsDefaultHost() (string, error) {
-	if len(cfg.APUrls) == 1 && cfg.APUrls[0].String() == DefaultInitialAdvertisePeerURLs {
-		return defaultHostname, defaultHostStatus
+func (cfg *Config) IsDefaultHost() (string, error) {
+	h, herr := netutil.GetDefaultHost()
+	if len(cfg.APUrls) == 1 {
```

Comment:
The 0.0.0.0 check should be on the listen url. If 0.0.0.0, then the advertise urls need to be checked if they match the default url before overriding.